### PR TITLE
Draft: Implement ast_builder Select::Order_by

### PR DESCRIPTION
**Pull request by woojinnn**
## Description
Implementation of [#696](https://github.com/gluesql/gluesql/issues/696)

## Review Required:
- [ ] FSM structure
  - Currently, fsm is structure as:
  - {Select, Having, GroupBy} -> {OrderBy} -> {Offset, Limit, Project}

## Remaining Works:  
from Issue 696
- [ ] Support In Sub query Expression
- [ ] Support null first

### CODE REVIEW

File: core/src/ast_builder/order_by_expr.rs [hunk 1]
```diff
@@ -1,29 +1,51 @@
...
+    fn try_from(order_by_expr_node: OrderByExprNode) -> Result<Self> {
+        match order_by_expr_node {
+            OrderByExprNode::OrderByExpr(order_by_expr) => Ok(order_by_expr),
+            OrderByExprNode::Text(order_by_expr) => {
+                parse_order_by_expr(order_by_expr).and_then(|op| translate_order_by_expr(&op))
+            }
+            OrderByExprNode::Expr(expr_node) => {
+                let expr = Expr::try_from(expr_node)?;
+
+                Ok(OrderByExpr { expr, asc: None })
```
Comment (24seconds) [inline core/src/ast_builder/order_by_expr.rs:48]: [Question]
Is it always `None`?

File: core/src/ast_builder/order_by_expr_list.rs [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    crate::{
+        ast::OrderByExpr,
+        parse_sql::parse_order_by_exprs,
+        result::{Error, Result},
+        translate::translate_order_by_expr,
+    },
+};
+
+#[derive(Clone)]
+pub enum OrderByExprList {
```
Comment (24seconds) [inline core/src/ast_builder/order_by_expr_list.rs:12]: [Question about this file]
Is it okay to skip writing test cases?

File: core/src/ast_builder/select/root.rs [hunk 3]
```diff
@@ -117,5 +123,9 @@ mod tests {
             .build();
         let expected = "SELECT * FROM Foo WHERE col1 > col2";
         test(actual, expected);
+
+        let actual = table("Foo").select().group_by("Country, Age").build();
```
Comment (24seconds) [inline core/src/ast_builder/select/root.rs:127]: [Question]
Is there a reason to add `group by` test case, not a `order by` test case?

File: core/src/ast_builder/select/order_by.rs [hunk 4]
```diff
@@ -0,0 +1,119 @@
+use {
+    super::{NodeData, Prebuild},
+    crate::{ast::Statement, ast_builder::*, result::Result},
```
Comment (24seconds) [inline core/src/ast_builder/select/order_by.rs:3]: [IMO, Suggestion]
I think gluesql does not use `*` for importing crate. How about replace `*` just like [group_by.rs](https://github.com/gluesql/gluesql/blob/3a604c93dddd9e4dcc47357ee3ecce22299709b0/core/src/ast_builder/select/group_by.rs#L5) ?

**Review by 24seconds — COMMENTED** [review]
Thank you for the tough work! Looks good to me overall.
I left some questions and comments. Could you take a look?

**Comment by 24seconds** [comment]
Oops. It's draft. @woojinnn is it still in draft stage?

**Comment by woojinnn** [comment]
> Oops. It's draft. @woojinnn is it still in draft stage?

Spent quite long time after commit. I'm not sure whether I'll continuously devote on this issue. Others may take this issue.

**Pull request closed without merging**